packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/service_account/provider.py

```
@@ -0,0 +1,136 @@
+    def _load_function(self, function_path: str) -> typing.Callable:
+        """Load a Python function from a module path string (e.g., 'my_module.get_token')."""
+        try:
+            module_name, func_name = function_path.rsplit(".", 1)
+            module = importlib.import_module(module_name)
+            func = getattr(module, func_name)
+            logger.info("Loaded service token function: %s", function_path)
+            return func
+        except Exception as e:
+            raise ValueError(f"Failed to load service token function '{function_path}': {e}") from e
+
+    async def authenticate(self, user_id: str | None = None, **kwargs) -> AuthResult:
```

Ruff (0.14.4)

Unused method argument: `user_id`

(ARG002)